rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -1656,156 +1630,6 @@ fn check_stream_handler_locally_generated_reject_response_unknown() {
     );
 }
 
-// TODO: [MR-579] Remove this fixture and the tests that depend on it, since reject responses of
-// this type are no longer produced.
-/// Common implementation for tests checking reject responses generated by the `StreamHandler`
-/// directly.
-fn check_stream_handler_generated_reject_response_impl(
```

Comment:
I put them back in and made them check for the reject signal, but I can't use `induct_loopback_stream()` for this directly, because that garbage collects the signals immediately. So I just induct the loopback stream as a slice instead.

For some signal types you could use `induct_loopback_stream()` because then we should see a reject response generated for the signal during the garbage collecting phase, but this is only true for some signals. For instance, with a canister migration, the reject response is routed into a stream instead and other constellations wouldn't even be possible at all such as `CanisterStopped` since there can't be a request in-flight from a canister that is stopped. What happens here is that the reject response is silently dropped.
You could also check for that but not in a unified way for all reject reasons and that's also not what these tests were supposed to test in the first place. Even before the tests were generating impossible constellations, except that the reject response ended up in the stream instead so the problem was circumvented.

Reply:
The old versions are also back in with a `legacy_` affix testing for `CertificationVersion::V18` and below.